types/shadow-dom.d.ts

```diff
@@ -0,0 +1,106 @@
+import { ChangeZoneEvent, InCombatChangedEvent, PlayerChangedEvent } from './events';
+
+export {};
+
+declare global {
+  type ShadowRootMode = 'open' | 'closed';
+
+  interface ShadowRootInit {
+    mode: ShadowRootMode;
+    delegatesFocus?: boolean;
+  }
+
+  interface ShadowRoot extends DocumentFragment, DocumentOrShadowRoot {
+    host: HTMLElement;
+    mode: ShadowRootMode;
+  }
+
+  interface Element {
+    attachShadow: (shadowRootInitDict: ShadowRootInit) => ShadowRoot;
+
+    shadowRoot: ShadowRoot | null;
+
```

Comment:
This is already declared in `libdom.d.ts`, you dont need to duplicate it.

Reply:
No, some properties are not. (I forget which ones, I'll reply after I finish `ui/test/test.ts`